client/context.go

```diff
@@ -220,6 +227,29 @@ func (ctx Context) WithInterfaceRegistry(interfaceRegistry codectypes.InterfaceR
 	return ctx
 }
 
+// WithViper returns the context with Viper field
+func (ctx Context) WithViper() Context {
+	v := viper.New()
+	ctx.Viper = v
+	return ctx
+}
+
+// WithHomeFlag checks if home flag is changed.
+// If this is a case, we update HomeDir field of Client Context
+/* Discovered a bug with Cory
+./build/simd init andrei --home ./test
+cd test/config there is no client.toml configuration file
+*/
+
+func (ctx Context) WithHomeFlag(cmd *cobra.Command) Context {
+	if cmd.Flags().Changed(flags.FlagHome) {
+		rootDir, _ := cmd.Flags().GetString(flags.FlagHome)
+		ctx = ctx.WithHomeDir(rootDir)
+	}
+
+	return ctx
+}
```

Comment:
All `With*` functions are simple setters which take the value. I don't like that we're mixing concerns here by importing cobra, we should just keep `WithHomeDir`.

Maybe the `if cmd.Flags().Changed(flags.FlagHome) {...}` logic could just be inlined in `simapp/simd/cmd/root.go.NewRootCmd`?

Reply:
Done